flows/runs/run.go

```diff
@@ -163,10 +163,29 @@ func (r *flowRun) SetWebhook(rr *utils.RequestResponse) {
 func (r *flowRun) Extra() utils.JSONFragment         { return r.extra }
 func (r *flowRun) SetExtra(extra utils.JSONFragment) { r.extra = extra }
 
+func (r *flowRun) ExpiresOn() *time.Time { return r.expiresOn }
+func (r *flowRun) ResetExpiration(from *time.Time) {
+	if r.Flow().ExpireAfterMinutes() != nil {
+		if from == nil {
+			now := time.Now().UTC()
+			from = &now
+		}
+
+		expiresAfterMinutes := time.Duration(*r.Flow().ExpireAfterMinutes())
+		expiresOn := from.Add(expiresAfterMinutes * time.Minute)
+
+		r.expiresOn = &expiresOn
+		r.setModifiedOn(time.Now().UTC())
```

Comment:
Removing those at the run level would remove the weirdness of parent and child run timestamps wouldn't it? bonus! I'm fine with either really.